solutions/fully-configurable/main.tf

```diff
@@ -32,3 +39,121 @@ module "scc_wp" {
   scc_wp_service_plan           = var.scc_workload_protection_service_plan
   cbr_rules                     = var.cbr_rules
 }
+
+########################################################################################################################
+# Cloud Security Posture Management (CSPM)
+########################################################################################################################
+
+resource "ibm_resource_instance" "app_configuration_instance" {
+  plan              = "basic"
+  name              = "${var.prefix}-conf-agg"
+  location          = var.region
+  resource_group_id = module.resource_group.resource_group_id
+  service           = "apprapp"
+}
+
+# Trusted Profile for Workload Protection
+resource "ibm_iam_trusted_profile" "workload_protection_profile" {
+  count = var.cspm_enabled ? 1 : 0
+  name  = "${var.prefix}-workload-protection-trusted-profile"
+}
+
+# Null resource to enable CSPM via CLI
+resource "null_resource" "enable_cspm" {
+  count = var.cspm_enabled ? 1 : 0
+
+  provisioner "local-exec" {
```

Comment:
As discussed can we look into using restapi provider here so that it doesn’t require an ibmcloud CLI session, and so we can support update / destroy as value will be stored in the state.
Make sure to leave a comment explaining why we do this (aka CSPM can only be enabled after the trusted profile exists, but profile can only exist after instance has been created, so hence we cannot enable it using `parameters_json` in the `ibm_resource_instance` block)